### DISCUSSION TITLE
Add Tft display

### DISCUSSION THREAD

**Discussion by Ahosan165** [risinek/esp32-wifi-penetration-tool#69] [Ideas]
It would be great if i could run it in a tft display connected with the esp32. Please make that possible. I tried my best to do that by myself but failed. Please,sir add the functionality of using this program form a tft display

**Comment by risinek** [comment]
Hi, sorry, this is completely out of scope of this project. One possible way would be to create a http client in the project that connects to the http server via localhost. But then it would require some lightweight browser engine to parse HTML and JS to display the UI. 

Another way is to create a UI and take inspiration from http server in the project how to bind UI to backend. I have never done that on ESP32, but here is some article from Espressif about the topic.
https://blog.espressif.com/making-the-fancy-user-interface-on-esp-has-never-been-easier-e44e79c0ae3

**Reply by Ahosan165** [reply]
Thanks for replying, i think i can make the ui using that software, but how will i intregate the ui with your code

**Reply by risinek** [reply]
see how the webserver component works. you basically need to call the same functions via UI.
For example the handler for starting attack 
https://github.com/risinek/esp32-wifi-penetration-tool/blob/e346e28ed9c93e98ab686c30eb8c4bc549313ba3/components/webserver/webserver.c#L120-L126
parses attack configuration into `attack_request_t` structure and then starts the attack by adding event into an event loop
https://github.com/risinek/esp32-wifi-penetration-tool/blob/e346e28ed9c93e98ab686c30eb8c4bc549313ba3/components/webserver/webserver.c#L124

so all you need is to call the `esp_event_post()` on pressing button for example. Feel free to try something and open PR. I will try to guide you through the integration (even though I have no experience with building UI on ESP32).

And to get data about the display, you do it the same way like here
https://github.com/risinek/esp32-wifi-penetration-tool/blob/e346e28ed9c93e98ab686c30eb8c4bc549313ba3/components/webserver/webserver.c#L144-L157
on this exact line, you get the `attack_status` structure that contains data about current attack progress.
https://github.com/risinek/esp32-wifi-penetration-tool/blob/e346e28ed9c93e98ab686c30eb8c4bc549313ba3/components/webserver/webserver.c#L147
